_posts/2017-03-06-new-topology-system.md

```diff
@@ -0,0 +1,102 @@
+---
+layout: post
+title: A shiny, new topology system 
+---
+
+With MDAnalysis 0.16.0 on the horizon, we wanted to showcase a major development that most users will probably not notice if we've done our job well.
+In fall 2015, @richardjgowers and I set to work on redesigning the topology system from scratch.
+This system determines how atom, residue, and segment information is internally represented and exposed to everything in the API (``Universe``, ``AtomGroup``, etc.), and the old scheme had issues with data duplication, maintaining consistency between atom and residue attributes, and performance for large systems.
+We hoped to resolve all of these issues with our new design.
+
+The starting point of this work was (the now infamous) [issue 363](https://github.com/MDAnalysis/mdanalysis/issues/363), which floated the idea of holding all atom, residue, and segment attributes in arrays instead of lists of ``Atom``, ``Residue``, and ``Segment`` objects.
+This approach turned the way topology data such as atom names, resids, masses, etc. are stored in a ``Universe`` on its head, going from an array of structs (list of ``Atom`` objects with individual attributes) to a struct of arrays (an array for each attribute, one entry per ``Atom``).
+
+Now, over a year later, the finishing touches on this work are being prepared for release.
+This post is meant to serve as a brief view to what has changed internally, what has changed externally, and what benefits this gives us looking forward to the future.
+
+## Internal changes that shouldn't affect external behavior
```

Comment:
Unlike @kain88-de I actually come to like starting with the internals better. However, you make it sound dull. Rather use a heading such as
```markdown
## Invisible changes that make working with MDAnalysis *faster*

Most of the changes are (or should be) invisible to the user. But they made some of the most fundamental operations in MDAnalysis quite a bit *faster*. Although this section is mostly of interest to developers, it is useful for all users to know the operations that MDAnalysis can now do much faster than before (and why).
```